core/coreapi/interface/path.go

```diff
@@ -125,6 +125,22 @@ func IpldPath(c cid.Cid) ResolvedPath {
 	}
 }
 
+// Simplify takes a ResolvedPath and returns a *new* ResovledPath that's as
+// short as possible. In the new resolved path, `p.Cid()` will be the same as
+// `p.Root()`.
+func Simplify(p ResolvedPath) ResolvedPath {
```

Comment:
Could be a method on the path object (`p.Simple()` would be a bit nicer IMO)

Reply:
Also, less stuff in `coreiface` namespace

Reply:
SGTM.